dot/peerset/peerstate.go

```diff
@@ -108,6 +109,8 @@ type PeersState struct {
 	// configuration of each set. The size of this Info is never modified.
 	// since, single Info can also manage the flow.
 	sets []Info
+
+	mu *sync.Mutex
```

Comment:
Does it have to be a pointer to a mutex? :thinking: A mutex value would do I think here